spring-integration-core/src/main/java/org/springframework/integration/endpoint/SourcePollingChannelAdapter.java

```diff
@@ -206,7 +227,12 @@ protected void handleMessage(Message<?> message) {
 
 	@Override
 	protected Message<?> receiveMessage() {
-		return this.source.receive();
+		if (this.limitingSource != null) {
+			return this.limitingSource.receive(this.maxFetchSize);
+		}
+		else {
+			return this.source.receive();
```

Comment:
Well, I wonder if we can live without `LimitingMessageSource` contract and just let the target `MessageSource` implementation to get deal with the provided `maxFetchSize` configuration.
Just like we have right now with the `JdbcPollingChannelAdapter`.

I just don't see reason to overhead `SourcePollingChannelAdapter` with the `maxFetchSize` logic, when it just doesn't provide any sense for that.

I mean we can add `maxFetchSize` to the `AbstractMessageSource` and that's all.
Although we may introduce different contract like:
```
interface MaxFetchSizeAware {
 
   void setMaxFetchSize();

}
```
or, of course, some other name.

But... From big height I can live with the current your solution, if you are not open to continue discussion 😄